src/overlays/actors/ovl_En_Zo/z_en_zo.c

```diff
@@ -15,17 +15,290 @@ void EnZo_Destroy(Actor* thisx, GlobalContext* globalCtx);
 void EnZo_Draw(Actor* thisx, GlobalContext* globalCtx);
 
-extern UNK_TYPE D_06000598;
-extern UNK_TYPE D_06000D48;
-extern UNK_TYPE D_060022F0;
-extern UNK_TYPE D_06002350;
-extern UNK_TYPE D_060024A0;
-extern UNK_TYPE D_06002510;
-extern UNK_TYPE D_0600BFA8;
-extern UNK_TYPE D_0600BFC0;
-extern UNK_TYPE D_0600C028;
+// Actions
+void EnZo_Standing(EnZo* this, GlobalContext* globalCtx);
+void EnZo_Submerged(EnZo* this, GlobalContext* globalCtx);
+void EnZo_Surface(EnZo* this, GlobalContext* globalCtx);
+void EnZo_TreadWater(EnZo* this, GlobalContext* globalCtx);
+void EnZo_Dive(EnZo* this, GlobalContext* globalCtx);
+
+extern Gfx D_060022F0[];
+extern Gfx D_06002350[];
+extern Gfx D_060024A0[];
+extern Gfx D_06002510[];
+extern Gfx D_0600BFC0[];
+extern Gfx D_0600C028[];
+extern FlexSkeletonHeader D_0600BFA8;
+
+extern AnimationHeader D_06002FE8;
+extern AnimationHeader D_06002F10;
+extern AnimationHeader D_0600219C;
+extern AnimationHeader D_06000598;
+extern AnimationHeader D_06000D48;
+
+// Surface Ripples
+void EnZo_Ripple(EnZo* this, Vec3f* pos, f32 scale, f32 targetScale, u8 alpha) {
+    EnZoEffect* effect;
+    Vec3f vec = { 0.0f, 0.0f, 0.0f };
+    s16 i;
+
+    effect = this->effects;
+    for (i = 0; i < ARRAY_COUNT(this->effects); i++) {
+        if (effect->type == 0) {
+            effect->type = 1;
+            effect->pos = *pos;
+            effect->scale = scale;
+            effect->targetScale = targetScale;
+            effect->color.a = alpha;
+            break;
+        }
+        effect++;
+    }
+}
+
+// Bubbles
+void EnZo_Bubble(EnZo* this, Vec3f* pos) {
+    EnZoEffect* effect;
+    Vec3f vec = { 0.0f, 0.0f, 0.0f };
+    Vec3f vel = { 0.0f, 1.0f, 0.0f };
+    s16 i;
+    f32 waterSurface;
+
+    effect = this->effects;
+    for (i = 0; i < ARRAY_COUNT(this->effects); i++) {
+        if (1) {}
+        if (effect->type == 0) {
+            waterSurface = this->actor.posRot.pos.y + this->actor.waterY;
+            if (!(waterSurface <= pos->y)) {
+                effect->type = 3;
+                effect->pos = *pos;
+                effect->vec = *pos;
+                effect->vel = vel;
+                effect->scale = ((Math_Rand_ZeroOne() - 0.5f) * 0.02f) + 0.12f;
+                break;
+            }
+        } 
+        effect++;
+    }
+}
+
+// Splashes
+void EnZo_Splash(EnZo* this, Vec3f* pos, Vec3f* vel, f32 scale) {
+    EnZoEffect* effect;
+    Vec3f accel = { 0.0f, -1.0f, 0.0f };
+    s16 i;
+
+    effect = this->effects;
+    for (i = 0; i < ARRAY_COUNT(this->effects); i++) {
+        if (1) {}
+        if (effect->type != 2) {
+            effect->type = 2;
+            effect->pos = *pos;
+            effect->vec = accel;
+            effect->vel = *vel;
+            effect->color.a = (Math_Rand_ZeroOne() * 100.0f) + 100.0f;
+            effect->scale = scale;
+            break;
+        } 
+        effect++;
+    }
+}
+
+void EnZo_Ripple_Fade(EnZo* this) {
+    EnZoEffect* effect = this->effects;
```

Comment:
we prefer to keep the function names `SystemName_FuncName` if possible without any extra underscores, unless providing a subsystem is necessary. Do you think it can be FadeRipple or something like that? 
Think this applies to a few other names in here